### DISCUSSION TITLE
Partitioned asset backfill and auto-materialization order

### DISCUSSION THREAD

**Discussion by littleK0i** [dagster-io/dagster#18511] [Q&A] [answered]
Currently when I start a backfill or auto-materialization for asset with many partitions, the execution order seems to be random. Reducing concurrency limit to 1 does not help to fix execution order.

In theory, this order should not make any difference, but quite often it does.

For example, if we update Snowflake table, the micro-partition layout could greatly benefit from sequential updates (1 Jan, 2 Jan, 3 Jan, ...). It helps to improve performance.

Also, if underlying asset is especially large, and each partition takes a long time to update, it would be much better if we could say say: "this table was updated up to XXX day". Currently we have to say: "~30% of days were updated in random order", which is not helpful.

Is there any way to enforce sequential order?

Thank you!

**Comment by clairelin135** [comment, marked as answer]
Hi @littleK0i. Unfortunately there isn't a built in way to enforce this ordering for asset backfills and auto-materialize, though one workaround would be to define a partition mapping like this, where each partition depends on the prior partition: https://github.com/dagster-io/dagster/discussions/11829 This would make Dagster respect the partition ordering.

If this behavior is insufficient/not entirely what you're looking for, please file a feature request.

**Reply by littleK0i** [reply]
Yes, it seems to be working perfectly. 👍 Not easy to set up, but gets the job done.

Thank you.

**Reply by luanmm** [reply]
Incremental assets is not an uncommon use-case. I am struggling to configure a graph to accomplish this. It could be easier, I think, and does not seems to be something that is too hard to implement, is it?
